src/expression.js

```diff
@@ -705,17 +705,24 @@ pp.parseFunctionBody = function(node, isArrowFunction) {
     // Start a new scope with regard to labels and the `inFunction`
     // flag (restore them to their old value afterwards).
     let oldLabels = this.labels
+    let oldVarDeclaredNames = this.varDeclaredNames
+    this.varDeclaredNames = new Set()
```

Comment:
(Or even arrays of strings + `indexOf`, which might be cheaper for small sets like this.)